src/kOS/Module/kOSProcessor.cs

```diff
@@ -298,15 +296,71 @@ private void InitUI()
             var options = (UI_ChooseOption)field.uiControlEditor;
 
-            var bootFiles = new List<string>();
+            var bootFiles = new List<VolumePath>();
 
-            bootFiles.Add("None");
             bootFiles.AddRange(BootDirectoryFiles());
 
             //no need to show the control if there are no available boot files
             options.controlEnabled = bootFiles.Count > 1;
             field.guiActiveEditor = bootFiles.Count > 1;
-            options.options = bootFiles.ToArray();
+            var availableOptions = bootFiles.Select(e => e.ToString()).ToList();
+            var availableDisplays = bootFiles.Select(e => e.Name).ToList();
+
+            availableOptions.Insert(0, "None");
+            availableDisplays.Insert(0, "None");
+
+            var bootFilePath = BootFilePath; // store the selected path temporarily
+            if (bootFilePath != null && !bootFiles.Contains(bootFilePath))
+            {
+                // if the path is not null ("None", null, or empty) and if it isn't in list of files
+                var archive = new Archive(SafeHouse.ArchiveFolder);
+                var file = archive.Open(bootFilePath) as VolumeFile;  // try to open the file as saved
+                if (file == null)
+                {
+                    // check the same file name, but in the boot directory.
+                    var path = VolumePath.FromString(BootDirectoryName).Combine(bootFilePath.Name);
+                    file = archive.Open(path) as VolumeFile; // try to open the new path
+                    if (file == null)
+                    {
+                        // try the file name without "boot" prefix
+                        var name = bootFilePath.Name;
+                        if (name.StartsWith("boot", StringComparison.OrdinalIgnoreCase) && name.Length > 4)
+                        {
+                            // strip the boot prefix and try that file name
+                            name = name.Substring(4);
+                            path = VolumePath.FromString(BootDirectoryName).Combine(name);
+                            file = archive.Open(path) as VolumeFile;  // try to open the new path
+                            if (file == null)
+                            {
+                                // try the file name without "boot_" prefix
+                                if (name.StartsWith("_", StringComparison.OrdinalIgnoreCase) && name.Length > 1)
+                                {
+                                    // only need to strip "_" here
+                                    name = name.Substring(1);
+                                    path = VolumePath.FromString(BootDirectoryName).Combine(name);
+                                    file = archive.Open(path) as VolumeFile;  // try to open the new path
+                                }
+                            }
+                        }
+                    }
+                }
```

Comment:
That's mostly correct.  Technically, including `0:/` in the path will break the boot file, as the `VolumePath` will simply parse that as another `/` separated segment.  

And with regard to the `boot[_]` prefix, the path can actually contain any additional information you desire, it only pays attention to the name of the file excluding the path information.  As this code is only called from within the editor, the only way to get a file name that is not either in the archive's root, or the archive's boot directory is to manually hack your craft file.  So a user can set `core:bootfilename` to `/boot/../libraries/orbit/../../boot/boot.ks` and this code would not interfere at all.  Adding `1:/` or `0:/` or `archive:/` would break the boot script.  If we want to make it valid to specify the volume of the boot script, we probably want to more drastically change the meaning of "start on archive".

In answer to your question though, yes, the behavior as you describe above is the intention.

You bring up a couple of potential documentation issues that I had not considered, and thus did not document.  Honestly, I don't know what happens if you use the `..` up directory shortcut on a volume root.  But other than the limitation of not specifying the volume, it should behave the same as any other path.